kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/health/KubernetesHealthIndicator.java

```diff
@@ -127,6 +120,13 @@ private static void accumulateContainerStatus(ArrayList<Object> list, ContainerS
 
     @Override
     protected Map<String, Object> getHealthInformation() {
+        try {
+            Pod pod = Single.fromPublisher(this.client.getPod(configuration.getNamespace(),
+                    System.getenv(HOSTNAME_ENV_VARIABLE))).blockingGet();
```

Comment:
Why blocking?

Reply:
To get latest container status. If this would be unblocking, it could happen that the old data are returned.

Reply:
Never mind, I forgot the method needs to return a map and not a flowable.

What you can in fact do is to cleanup unused variables:

- `executorService` is unused.
- The attribute `healthInformation` can be removed, and make `processPod` return the result.

Reply:
yes, that's true, missed that.. thanks for noticing

Reply:
as said, once we will have support for watching of the resources, we should replace this so it doesn't generate unnecessary traffic